src/sevcore_linux.cpp

```diff
@@ -771,17 +771,19 @@ int SEVDevice::generate_vcek_ask(const std::string output_folder,
     return cmd_ret;
 }
 
-int SEVDevice::request_platform_status(snp_platform_status_buffer &plat_status) {
-    int ioctl_return = 0;
-    snp_platform_status_cmd_buf buf = { .status_p_addr = (uint64_t)&plat_status };
-    sev_ioctl(SEV_SNP_PLATFORM_STATUS, &buf, &ioctl_return);
+int SEVDevice::request_platform_status(snp_platform_status_buffer *plat_status) {
+    int cmd_ret = SEV_RET_UNSUPPORTED;
+
+    memset(plat_status, 0, sizeof(snp_platform_status_buffer));
```

Comment:
It is my understanding that it is usually bad practice to use memset to zero-fill an object. Perhaps you mean to use `std::fill`?

Reply:
It is already used 38 times in the project, while `std::fill` is not used at all
```
❯ git grep memset | wc -l
38
```

Reply:
You are right! I will make a note to look into addressing this, as using `memset()` on a non-primitive may cause undefined behavior.